### DISCUSSION TITLE
Difference in working of override when using glob pattern

### DISCUSSION THREAD

**Discussion by anto-christo** [eslint/eslint#15515] [Q&A]
Hi, 

I am using the ESLint Node.js API.

```js
const eslint = new ESLint(api_config)
const report = await eslint.lintFiles(data.files) // data.files is an array of file paths
```

I pass the ESLint class a config that contains an override.
I am using globstar (**) to recursively match the files and apply this override to turn off certain rules.
However, the override is not working in this case.
```json5
// Not working
[
  {
    "files": [
      "**/*.test.js",
      "**/*.test.ts",
      "**/*.spec.js",
      "**/*.spec.ts",
      "**/*.test.cjs",
      "**/*.spec.cjs"
    ],
    "rules": {
      "no-console": "off",
      "no-debugger": "off",
      "no-undef": "off",
      "no-redeclare": "off"
    }
  }
]
```

When using only `*.{ext}`, the files are matched recursively and the override works as expected.
```json5
// Working!!
[
  {
    "files": [
      "*.test.js",
      "*.test.ts",
      "*.spec.js",
      "*.spec.ts",
      "*.test.cjs",
      "*.spec.cjs"
    ],
    "rules": {
      "no-console": "off",
      "no-debugger": "off",
      "no-undef": "off",
      "no-redeclare": "off"
    }
  }
]
```

Can someone please help me understand why the first scenario does not work and the second one does?

Best Regards,
Anto

**Comment by nzakas** [comment]
Can you explain what you mean by “doesn’t work”? What is it you are seeing? What are the actual file names that are being matched or not matched?

**Reply by anto-christo** [reply]
Hey Nicholas,

By "doesn't work" I mean the file is not matched and hence the override is not respected.
For example, I have a file with path `test/e2e/specs/index.spec.js`.

Above are the two snippets that I tried with the overrides array.

With the first snippet, this file is not matched and the override is not applied.
However, with the second snippet, the file is matched and the override is applied.

I am wondering, shouldn't `**/*.spec.js` in the first snippet match the above path?

**Reply by nzakas** [reply]
It should, however, that pattern is relative to the cwd. Double check that the pattern makes sense relative to your cwd, or else you can pass a `cwd` option to the ESLint constructor.
